src/javascript/app_2/Modules/Trading/Components/Form/ContractType/contract_type_widget.jsx

```diff
@@ -20,27 +24,82 @@ class ContractTypeWidget extends React.PureComponent {
         document.removeEventListener('mousedown', this.handleClickOutside);
     }
 
-    handleSelect = (item) => {
+    handleSelect = (item, e) => {
+        if (item.value !== this.props.value && e.target.id !== 'info-icon') {
+            this.props.onChange({ target: { name: this.props.name, value: item.value } });
+        }
+        this.handleVisibility();
+    };
+
+    onSubmitButtonClick = (item) => {
         if (item.value !== this.props.value) {
             this.props.onChange({ target: { name: this.props.name, value: item.value } });
         }
+        this.handleInfoVisibility();
+    };
+
+    handleInfoClick = (item) => {
+        this.setState({ item });
+        this.handleInfoVisibility();
         this.handleVisibility();
     };
 
+    handlePaginationClick = (item) => {
+        this.setState({ item });
+    };
+
+    handleNextClick = (paginationList) => {
+        const paginationLength = paginationList.length;
+        const item = this.state.item;
+        const currentIndex = paginationList.findIndex((list_item) => list_item.value === item.value);
+        const nextIndex = currentIndex + 1;
+        if (nextIndex < paginationLength) {
+            this.handlePaginationClick(paginationList[nextIndex]);
+        } else {
+            this.handlePaginationClick(paginationList[0]);
+        }
+    };
+
+    handlePrevClick = (paginationList) => {
+        const paginationLength = paginationList.length;
+        const item = this.state.item;
+        const currentIndex = paginationList.findIndex((list_item) => list_item.value === item.value);
+        const prevIndex = currentIndex - 1;
+        if (prevIndex > -1) {
+            this.handlePaginationClick(paginationList[prevIndex]);
+        } else {
+            this.handlePaginationClick(paginationList[paginationLength - 1]);
+        }
+    };
+
     setWrapperRef = (node) => {
         this.wrapper_ref = node;
     };
 
     handleClickOutside = (event) => {
         if (this.wrapper_ref && !this.wrapper_ref.contains(event.target) && this.state.is_dialog_open) {
             this.setState({ is_dialog_open: false });
+        } else if (this.wrapper_ref && !this.wrapper_ref.contains(event.target) && this.state.is_info_dialog_open) {
+            this.setState({ is_info_dialog_open: false, is_dialog_open: true });
         }
     };
 
+    handleInfoVisibility = () => {
+        this.setState({ is_info_dialog_open: !this.state.is_info_dialog_open });
```

Comment:
setState can be async because it can be batched so if you are changing state based on current state I think it should be like this I think?
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous
```
this.setState((state) => ({
 is_info_dialog_open: !state.is_info_dialog_open
}));
```